File: src/app/security/hashing.py
"""Hash do ``client_secret``.

O secret em texto puro só existe no instante em que é gerado e devolvido ao
administrador; o banco guarda exclusivamente o hash Argon2id.
"""

from __future__ import annotations

from argon2 import PasswordHasher
from argon2.exceptions import Argon2Error, InvalidHashError


class HasherSecret:
    """Fachada sobre o Argon2id com verificação que nunca levanta exceção."""

    def __init__(self, hasher: PasswordHasher | None = None) -> None:
        self._hasher = hasher or PasswordHasher()

    def gerar_hash(self, secret: str) -> str:
        return self._hasher.hash(secret)

    def verificar(self, secret: str, hash_armazenado: str) -> bool:
        """Compara o secret com o hash. Retorna ``False`` em qualquer falha."""
        try:
            return self._hasher.verify(hash_armazenado, secret)
        except (Argon2Error, InvalidHashError, TypeError, ValueError):
            return False

    def precisa_rehash(self, hash_armazenado: str) -> bool:
        try:
            return self._hasher.check_needs_rehash(hash_armazenado)
        except (InvalidHashError, TypeError, ValueError):
            return False


hasher_padrao = HasherSecret()

File: src/modules/hashing/hashing.ts
/**
 * Hash do `client_secret`.
 *
 * O secret em texto puro só existe no instante em que é gerado e devolvido
 * ao administrador; o banco guarda exclusivamente o hash Argon2id.
 */

import * as argon2 from 'argon2';

export async function gerarHash(secret: string): Promise<string> {
  return argon2.hash(secret);
}

/** Compara o secret com o hash. Retorna `false` em qualquer falha. */
export async function verificarHash(secret: string, hashArmazenado: string): Promise<boolean> {
  try {
    return await argon2.verify(hashArmazenado, secret);
  } catch {
    return false;
  }
}

export async function precisaRehash(hashArmazenado: string): Promise<boolean> {
  try {
    return argon2.needsRehash(hashArmazenado);
  } catch {
    return false;
  }
}
